### ISSUE TITLE
Haml duplicates content with turbo_stream and if statement

### ISSUE THREAD

**Issue by 0llirocks** [haml/haml#1169]
Rails: 7.2.1
Haml: 6.3.0

controller.rb

```
def index
    respond_to do |format|
      format.html
      format.turbo_stream
    end
end
```

index.turbo_stream.haml

```
Test
= if true
  Test
```

/index.turbo_stream returns

```
Test
Test
Test
Test
```

If I change index.turbo_stream.haml to if false everything works

index.turbo_stream.haml

```
Test
= if false
  Test
```

/index.turbo_stream returns

```
Test
```

I already changed to .erb instead of .haml and everything works as expected, I assume it's a haml issue.

**Comment by jeffpeterson** [comment]
I think you need to use `-` instead of `=`:

```
- if true
  Test
```

The content inside the body of the if statement already gets added to the output buffer. `=` adds it again, but `-` does not.

**Comment by 0llirocks** [comment]
Good point, you are correct. I will verify if this fixes to error. But why is it printed 4x times? Shouldn't it be just 3 then?
